src/contract/helpers.js

```diff
@@ -0,0 +1,158 @@
+const { ethers } = require('ethers');
+const config = require('../config');
+const constants = require('../lib/constants');
+const StakingPoolFactoryAbi = require('../abis/StakingPoolFactory.json');
+const StakingPoolAbi = require('../abis/StakingPool.json');
+const CoverAbi = require('../abis/Cover.json');
+const StakingViewerAbi = require('../abis/StakingViewer.json');
+const StakingProductAbi = require('../abis/StakingProducts.json');
+
+const { getCreate2Address } = ethers.utils;
+const { INIT_CODE_HASH, CONTRACTS_ADDRESSES, STAKING_POOL_STARTING_ID } = constants;
+const url = config.get('provider.http');
+const provider = new ethers.providers.JsonRpcProvider(url);
+
+function calculateCurrentTrancheId() {
+  return Math.floor(Date.now() / (constants.TRANCHE_DURATION_DAYS * 24 * 3600 * 1000));
+}
+
+function calculateAddress(id) {
+  const salt = Buffer.from(id.toString(16).padStart(64, '0'), 'hex');
+  const initCodeHash = Buffer.from(INIT_CODE_HASH, 'hex');
+  return getCreate2Address(CONTRACTS_ADDRESSES.StakingPoolFactory, salt, initCodeHash);
+}
+
+/* ================== STAKING POOLS ================================== */
+
+async function fetchStakingPoolIdAndAddress() {
+  const StakingPoolFactory = new ethers.Contract(
+    CONTRACTS_ADDRESSES.StakingPoolFactory,
+    StakingPoolFactoryAbi,
+    provider,
+  );
+
+  const count = await StakingPoolFactory.stakingPoolCount();
+
+  const pools = [];
+  for (let id = STAKING_POOL_STARTING_ID; id < STAKING_POOL_STARTING_ID + Number(count); id += 1) {
```

Comment:
They'll definitely start at 1. Can safely use `for (let i = 1; i <= count.toNumber(); i++)`